### PR TITLE
STD support for PSVita

### PR DESCRIPTION
**Pull request by nikarh**
This PR adds std support for `armv7-sony-vita-newlibeabihf` target.

The work here is fairly similar to #95897, just for a different target platform. 

This depends on the following pull requests:

rust-lang/backtrace-rs#523
rust-lang/libc#3209

### CODE REVIEW

File: src/doc/rustc/src/platform-support/armv7-sony-vita-newlibeabihf.md [hunk 1]
```diff
@@ -1,26 +1,49 @@
...
+
 ## Designated Developers
 
 * [@amg98](https://github.com/amg98)
+* [@nikarh](https://github.com/nikarh)
 
 ## Requirements
 
-This target is cross compiled, and requires installing [VITASDK](https://vitasdk.org/) toolchain on your system.
+This target is cross-compiled, and requires installing oolchain on your system.. Dynamic linking is not supported.
```
[the remark this answers is not in the document]
Comment (nikarh) [reply src/doc/rustc/src/platform-support/armv7-sony-vita-newlibeabihf.md]: Oops, fixed.

**Comment by nikarh** [comment]
Sorry for the force-push spam.

rust-lang/libc#3209 is merged, rust-lang/backtrace-rs#523 is waiting for review.

I would rather wait until `libc` is released and `backtrace-rs` is merged, so I'll convert this PR to draft for now.

**Comment by nikarh** [comment]
Now that libc is released, waiting for https://github.com/rust-lang/backtrace-rs/pull/523 to be merged (I assume CI has to be fixed first for rust nightly in its master, but that's out of the scope of my PRs).

After that, I'll update and un-draft this PR.

**Comment by Mark-Simulacrum** [comment]
Please squash commits, r=me with that done.

**Comment by BlackSheepBoy69** [comment]
Does it output as a .vpk file or an eboot.bin? Do you need VitaSDK to use this for Vita?

**Comment by nikarh** [comment]
> Does it output as a .vpk file or an eboot.bin? Do you need VitaSDK to use this for Vita?

All relevant doc is here:
https://doc.rust-lang.org/nightly/rustc/platform-support/armv7-sony-vita-newlibeabihf.html

tldr; you can create a static lib in Rust, link it with VitaSDK to an elf, and then do the rest up to vpk with VitaSDK tools.
If you need any help, I suppose it's better to contact me directly, instead of here.

**Comment by bigskribbz** [comment]
Better put a condom on that 👀
